services/management/ia/rules_service.go

```diff
@@ -355,6 +352,87 @@ func (s *RulesService) CreateAlertRule(ctx context.Context, req *iav1beta1.Creat
 	return &iav1beta1.CreateAlertRuleResponse{RuleId: rule.ID}, nil
 }
 
+func (s *RulesService) processRuleParameters(param []*iav1beta1.RuleParam, templateName string) (models.RuleParams, error) {
+	ruleParams, err := convertRuleParamsToModel(param)
+	if err != nil {
+		return nil, err
+	}
+
+	t, ok := s.templates.getTemplates()[templateName]
+	if !ok {
+		return nil, status.Errorf(codes.NotFound, "Unknown template %s.", templateName)
+	}
+
+	unknownParams := make(map[string]bool)
+
+	res := make(models.RuleParams, 0, len(t.Params))
+	for _, tp := range t.Params {
+		var filled bool
+		for _, rp := range ruleParams {
+			if _, ok := unknownParams[rp.Name]; !ok {
```

Comment:
for better readability we can extract filling map from the loop whole loop and instead of setting `false` as value just delete the pair from map.

Reply:
Agree, done